### ISSUE TITLE
Getting Nan's on model-parallel

### ISSUE THREAD

**Issue by adizhol** [pytorch/pytorch#34134]
Hi,

I've been trying to train a maskcrnn-benchmark with backbones of Resnet-FPN and ResneXt-FPN on multiple GPUs (V100) using model-parallel approach as described in https://pytorch.org/tutorials/intermediate/model_parallel_tutorial.html.

After a few iterations both model encounter Nan's in backbone output.
This problem doesn't occur when training on a single GPU (same learning rate).

pytorch 1.2
cuda 10.1.168

Has anyone encountered this problem?

Thanks

**Comment by ptrblck** [comment]
Could you update to the latest stable release (`1.4.`) and retry our code?
Also, could you post a code snippet to reproduce this issue?

**Comment by mrshenli** [comment]
Hey @adizhol, could you please share a minimum repro for this problem? It could go wrong in various ways, e.g., streams not synchronized properly, accumulated batch size are different model-parallel vs non-model-parallel runs, etc.

**Comment by adizhol** [comment]
@ptrblck @mrshenli 
I don't have a public repo, but basically, my code is like this:
initializing trainer:
```
    device_ids = cfg.MODEL.DEVICE.split(',') # cuda:0,cuda:1
    model = build_detection_model(cfg)
    if len(device_ids) > 1: #keep backbone in device 0. The rest go to device 1
        model.to(device_ids[1])
        model.backbone.body = model.backbone.body.to(device_ids[0])
        model.backbone.fpn = model.backbone.fpn.to(device_ids[1])
        model.rpn = model.rpn.to(device_ids[1])
    else:
        model.to(device)
```
The forward of maskrcnn:
```
        if len(self.devices) > 1:
            resnet_out = self.backbone.body(images.tensors) #inputs are in device 0
            features = self.backbone.fpn([x.to(self.devices[1]) for x in resnet_out])
        else:
            features = self.backbone(images.tensors)
        proposals, proposal_losses = self.rpn(images, features, targets)
        if self.roi_heads:
            x, result, detector_losses = self.roi_heads(features, proposals, targets) #non Nans up to here
...
```
There are no Nans until calling the roi_head

**Comment by adizhol** [comment]
update: The Nan's are from the poolers, probably from ROIAlign...
Any idea if there special care needs to be taken in .cu code to support model-parallel?

**Comment by adizhol** [comment]
update: upgrading to 1.4 did not solve the problem

**Comment by adizhol** [comment]
solved by replacing ROIALIGN with torchvision's ROIAlign
